noodles-editor/src/noodles/store.tsx

```diff
@@ -1,91 +1,142 @@
 import type { ISheetObject } from '@theatre/core'
-import { createContext, type PropsWithChildren, useContext, useMemo } from 'react'
 import { create } from 'zustand'
 import type { IOperator, Operator } from './operators'
 // only import types from noodles to avoid circular dependencies
 import type { OpId } from './utils/id-utils'
 import { isAbsolutePath, resolvePath } from './utils/path-utils'
 
-export const opMap = new Map<OpId, Operator<IOperator>>()
-export const sheetObjectMap = new Map<OpId, ISheetObject>()
-
 // ============================================================================
-// Nesting State (Zustand)
+// Operator Store (Zustand)
 // ============================================================================
 
-interface NestingState {
-  currentContainerId: string
-  setCurrentContainerId: (id: string) => void
+interface OperatorStore {
+  // The actual maps
+  operators: Map<OpId, Operator<IOperator>>
+  sheetObjects: Map<OpId, ISheetObject>
+  hoveredOutputHandle: { nodeId: string; handleId: string } | null
+
+  // Version counter for change tracking
+  version: number
+
+  // Batching state
+  _batching: boolean
+  _pendingVersion: number
+
+  // Operator actions
+  getOp: (id: OpId) => Operator<IOperator> | undefined
+  setOp: (id: OpId, op: Operator<IOperator>) => void
+  deleteOp: (id: OpId) => void
+  hasOp: (id: OpId) => boolean
+  clearOps: () => void
+  getAllOps: () => Operator<IOperator>[]
+  getOpEntries: () => [OpId, Operator<IOperator>][]
+
+  // Sheet object actions
+  getSheetObject: (id: OpId) => ISheetObject | undefined
+  setSheetObject: (id: OpId, sheetObj: ISheetObject) => void
+  deleteSheetObject: (id: OpId) => void
+  hasSheetObject: (id: OpId) => boolean
+
+  // Hovered output handle actions
+  setHoveredOutputHandle: (handle: { nodeId: string; handleId: string } | null) => void
+
+  // Batching
+  batch: (fn: () => void) => void
 }
 
-export const useNestingStore = create<NestingState>((set) => ({
-  currentContainerId: '/',
-  setCurrentContainerId: (id: string) => set({ currentContainerId: id }),
-}))
+export const useOperatorStore = create<OperatorStore>((set, get) => ({
+  operators: new Map(),
+  sheetObjects: new Map(),
+  hoveredOutputHandle: null,
+  version: 0,
```

Comment:
What would version be useful for?

Reply:
I think this was more to do with adding a history entry, but then it's not as important now. I can remove